src/sys/prctl.rs

```
@@ -0,0 +1,229 @@
+
+    let res = unsafe {
+        libc::prctl(PR_GET_CHILD_SUBREAPER as c_int, subreaper_ptr as c_ulong, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|_| subreaper != 0)
+}
+
+/// Set the dumpable attribute which determines if core dumps are created for this process.
+pub fn set_dumpable(attribute: bool) -> Result<()> {
+    prctl_set_bool(PR_SET_DUMPABLE, attribute)
+}
+
+/// Get the dumpable attribute for this process.
+pub fn get_dumpable() -> Result<bool> {
+    let res = unsafe {
+        libc::prctl(PR_GET_DUMPABLE as c_int, 0, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|res| res != 0)
+}
+
+/// Set the "keep capabilities" attribute for this process. This causes the thread to retain
+/// capabilities even if it switches its UID to a nonzero value.
+pub fn set_keepcaps(attribute: bool) -> Result<()> {
+    prctl_set_bool(PR_SET_KEEPCAPS, attribute)
+}
+
+/// Get the "keep capabilities" attribute for this process
+pub fn get_keepcaps() -> Result<bool> {
+    let res = unsafe {
+        libc::prctl(PR_GET_KEEPCAPS as c_int, 0, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|res| res != 0)
+}
+
+/// Clear the thread memory corruption kill policy and use the system-wide default
+pub fn clear_mce_kill() -> Result<()> {
+    let res = unsafe {
+        libc::prctl(PR_MCE_KILL as c_int, libc::PR_MCE_KILL_CLEAR, 0, 0, 0)
+    };
+
+    Errno::result(res).map(drop)
+}
+
+/// Set the thread memory corruption kill policy
+pub fn set_mce_kill(policy: PrctlMCEKillPolicy) -> Result<()> {
+    let res = unsafe {
+        libc::prctl(PR_MCE_KILL as c_int, libc::PR_MCE_KILL_SET, policy as c_ulong, 0, 0)
+    };
+
+    Errno::result(res).map(drop)
+}
+
+/// Get the thread memory corruption kill policy
+pub fn get_mce_kill() -> Result<PrctlMCEKillPolicy> {
+    let res = unsafe {
+        libc::prctl(PR_MCE_KILL_GET as c_int, 0, 0, 0, 0)
+    };
+
+    Errno::result(res).map(|res| PrctlMCEKillPolicy::try_from(res).unwrap())
```

This unwrap could panic in the future if the kernel adds a new kill policy.  Better to return an error in that case.

---

Fixed it :+1: